torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -901,26 +913,30 @@ struct to_ir {
   }
 
   void emitDelete(const Delete& stmt) {
-    if (stmt.expr().kind() != TK_SUBSCRIPT) {
-      throw ErrorReport(stmt.range())
-          << "del statements are only supported for list"
-             " and dict item deletion";
-    }
-    Subscript subscript(stmt.expr());
-    const List<Expr>& subscript_exprs = subscript.subscript_exprs();
-    if (subscript_exprs[0].kind() == TK_SLICE_EXPR) {
+    if (stmt.expr().kind() == TK_SUBSCRIPT) {
+      Subscript subscript(stmt.expr());
+      const List<Expr>& subscript_exprs = subscript.subscript_exprs();
+      if (subscript_exprs[0].kind() == TK_SLICE_EXPR) {
+        throw ErrorReport(stmt.range())
+            << "del statements only support deletion at a single index, "
+               "slicing is not supported"
+               " (see https://github.com/pytorch/pytorch/issues/31430)";
+      }
+      const SugaredValuePtr sv = emitSugaredExpr(subscript.value(), 1);
+      const SourceRange& val_range = subscript.value().range();
+      Value* idx = emitExpr(subscript_exprs[0]);
+      Value* val = sv->asValue(val_range, method);
+      auto node = graph->create(aten::Delete, {val, idx}, 0)
+                      ->setSourceRange(stmt.range());
+      graph->insertNode(node);
+    } else if (stmt.expr().kind() == TK_VAR) {
+      Var var(stmt.expr());
+      auto n = environment_stack->removeVar(var.name().name());
```

Comment:
why is `n` being assigned here? 

I think we probably want to call `getVar` before we call `removeVar` here so that an error is thrown if the variable isn't defined.

Reply:
Oops, forgot to add an assert afterward. But if adding a call to getVar is preferred, I'll do that instead, lemme know.

Reply:
Either is fine, but if you don't do `getVar` please factor out this part of the code https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/frontend/ir_emitter.cpp#L512